src/applications/vaos/new-appointment/redux/helpers/formSubmitTransformers.js

```diff
@@ -127,27 +128,43 @@ export function transformFormToVARequest(state) {
 
 export function transformFormToCCRequest(state) {
   const data = getFormData(state);
+  const useProviderSelection = selectUseProviderSelection(state);
+  const provider = data.communityCareProvider;
   let preferredProviders = [];
 
-  if (data.hasCommunityCareProvider) {
-    const street = `${data.communityCareProvider.address.street}, ${
-      data.communityCareProvider.address.street2
-    }`;
+  if (
+    useProviderSelection &&
+    !!data.communityCareProvider &&
+    Object.keys(data.communityCareProvider).length
+  ) {
+    preferredProviders = [
+      {
+        address: {
```

Comment:
I looked in the Swagger for this api, it looks like the `address` object is expected and the extra `providerStreet` etc fields are not, so I left them out.

https://coderepo.mobilehealth.va.gov/projects/VAR/repos/var-resources/browse/var-resources/src/main/resources/swagger.json#5068